File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueCommentUpdatedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.IssueComment;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

public record IssueCommentUpdatedEvent(
        long projectId,
        long issueId,
        long commentId,
        long editorId,
        Instant updatedAtUtc
) {
    public static IssueCommentUpdatedEvent from(IssueComment comment, long projectId, long editorId) {
        LocalDateTime commentUpdatedAt = comment.getUpdatedAt();
        Instant updatedAtInstant = (commentUpdatedAt != null)
                ? commentUpdatedAt.atZone(ZoneId.of("UTC")).toInstant()
                : Instant.now();

        return new IssueCommentUpdatedEvent(
                projectId,
                comment.getIssue().getId(),
                comment.getId(),
                editorId,
                updatedAtInstant
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueCommentUpdatedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueCommentUpdatedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("commentId")]
    public long CommentId { get; set; }

    [JsonPropertyName("editorId")]
    public long EditorId { get; set; }

    [JsonPropertyName("updatedAtUtc")]
    public DateTimeOffset UpdatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}